components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultAuthenticationRequestHandler.java

```diff
@@ -1057,13 +1058,26 @@ private void addAuditLogs(String sessionAction, String authenticatedUser, String
         JSONObject auditData = new JSONObject();
         auditData.put(SessionMgtConstants.SESSION_CONTEXT_ID, sessionKey);
         auditData.put(SessionMgtConstants.REMEMBER_ME, isRememberMe);
-        auditData.put(SessionMgtConstants.AUTHENTICATED_USER, authenticatedUser);
         auditData.put(SessionMgtConstants.AUTHENTICATED_USER_TENANT_DOMAIN, userTenantDomain);
         auditData.put(SessionMgtConstants.TRACE_ID, traceId);
+
+        String initiator;
+        if (LoggerUtils.isLogMaskingEnable) {
+            String maskedUsername = LoggerUtils.maskContent(authenticatedUser);
+            auditData.put(SessionMgtConstants.AUTHENTICATED_USER, maskedUsername);
+            if (StringUtils.isNotBlank(userTenantDomain)) {
+                initiator = IdentityUtil.getInitiatorId(authenticatedUser, userTenantDomain);
```

Comment:
in ApplicationMgtAuditLogger.java you have also checked if the initiator is not blank. better to be consistent